junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/DiscoverySelectors.java

```diff
@@ -282,20 +287,24 @@ public static JavaClassSelector selectJavaClass(String className) {
 	 * <tr><td>{@link String#substring(int, int)}</td><td>{@code java.lang.String#substring(int, int)}</td></tr>
 	 * </table>
 	 *
-	 * @param name the fully qualified name of the method to select; never
+	 * @param methodName the fully qualified methodName of the method to select; never
 	 * {@code null} or blank
-	 * @throws PreconditionViolationException if the supplied name is {@code null},
+	 * @throws PreconditionViolationException if the supplied methodName is {@code null},
 	 * blank, or does not specify a unique method
 	 * @see JavaMethodSelector
 	 */
-	public static JavaMethodSelector selectJavaMethod(String name) throws PreconditionViolationException {
-		Preconditions.notBlank(name, "Method name must not be null or blank");
+	public static JavaMethodSelector selectJavaMethod(String methodName) throws PreconditionViolationException {
+		Preconditions.notBlank(methodName, "Method name must not be null or blank");
 
-		Optional<Method> methodOptional = ReflectionUtils.loadMethod(name);
-		Method method = methodOptional.orElseThrow(() -> new PreconditionViolationException(
-			String.format("'%s' could not be resolved to a unique method", name)));
+		Matcher matcher = fullyQualifiedMethodNamePattern.matcher(methodName);
+		Preconditions.condition(matcher.matches(), "Method name is not a valid full qualified method name");
 
-		return selectJavaMethod(method.getDeclaringClass(), method);
+		if (StringUtils.isNotBlank(matcher.group(3))) {
```

Comment:
I would prefer to have the various `matcher.group(index)`-calls assigned to variables with explanatory names for better readability